prow/plugins/blockade/blockade.go

```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Example config for this plugin:
+
+blockades:
+- repos:
+	- kubernetes-incubator
+	- kubernetes/kubernetes
+	- kubernetes/test-infra
+  blockregexps:
+  - "docs/.*"
+  - "other-docs/.*"
+  exceptionregexps:
+  - ".*OWNERS"
+  explanation: "Files in the 'docs' directory should not be modified except for OWNERS files"
+- repos:
+  - kubernetes/test-infra
+  blockregexps:
+  - "mungegithub/.*"
+  exceptionregexps:
+  - "mungegithub/DeprecationWarning.md"
+  explanation: "Don't work on mungegithub! Work on Prow!"
+*/
+
+package blockade
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const (
+	pluginName        = "blockade"
+	blockedPathsLabel = "do-not-merge/blocked-paths"
```

Comment:
Should this be part of the config? Bad docs changes had a special label before. Other users might want a different label to be applied.

Reply:
If we let each blockade specify its own label and a blockade is removed, we won't know about that blockade's label anymore and won't be able to remove it from PRs that already have it. 
We could allow partial customization of the label by allowing blockades to specify a label suffix and prefix all labels with `do-not-merge/blockade/` or something like that. Then the plugin can remove any labels with the prefix that no longer apply. That would result in long label names and additional label bloat though.